rasa/nlu/featurizers/sparse_featurizer/regex_featurizer.py

```diff
@@ -160,23 +291,60 @@ def load(
         model_dir: Optional[Text] = None,
         model_metadata: Optional[Metadata] = None,
         cached_component: Optional["RegexFeaturizer"] = None,
+        should_finetune: bool = False,
         **kwargs: Any,
     ) -> "RegexFeaturizer":
-
+        """Loads a previously trained component.
+
+        Args:
+            meta: Configuration of trained component.
+            model_dir: Path where trained pipeline is stored.
+            model_metadata: Metadata for the trained pipeline.
+            cached_component: Previously cached component(if any).
+            should_finetune: Indicates whether to load the component for further finetuning.
+            **kwargs:
+        """
         file_name = meta.get("file")
-        regex_file = os.path.join(model_dir, file_name)
 
-        if os.path.exists(regex_file):
-            known_patterns = rasa.shared.utils.io.read_json_file(regex_file)
-            return RegexFeaturizer(meta, known_patterns=known_patterns)
-        else:
-            return RegexFeaturizer(meta)
+        patterns_file_name = Path(model_dir).joinpath(file_name + ".patterns.pkl")
+
+        vocabulary_stats_file_name = Path(model_dir).joinpath(
+            file_name + ".vocabulary_stats.pkl"
+        )
+
+        known_patterns = None
+        vocabulary_stats = None
+        if patterns_file_name.exists():
+            known_patterns = rasa.shared.utils.io.read_json_file(
+                str(patterns_file_name)
+            )
+        if vocabulary_stats_file_name.exists():
+            vocabulary_stats = rasa.shared.utils.io.read_json_file(
+                str(vocabulary_stats_file_name)
+            )
+
+        return RegexFeaturizer(
+            meta,
+            known_patterns=known_patterns,
+            pattern_vocabulary_stats=vocabulary_stats,
+            finetune_mode=should_finetune,
+        )
 
     def persist(self, file_name: Text, model_dir: Text) -> Optional[Dict[Text, Any]]:
         """Persist this model into the passed directory.
-        Return the metadata necessary to load the model again."""
-        file_name = file_name + ".pkl"
-        regex_file = os.path.join(model_dir, file_name)
-        utils.write_json_to_file(regex_file, self.known_patterns, indent=4)
+
+        Args:
+            file_name: Prefix to add to all files stored as part of this component.
+            model_dir: Path where files should be stored.
+
+        Returns:
+            Metadata necessary to load the model again.
+        """
+        patterns_file_name = file_name + ".patterns.pkl"
+        regex_file = Path(model_dir).joinpath(patterns_file_name)
```

Comment:
Isn't that Linux specific then because we'll use `/` and would fail for Windows? Or does `Path` take care of that?

Reply:
always used `/` so far and never had problems 🤔 so I guess `Path` takes care of that.

Reply:
I do like the descriptive and non-ambiguous nature of using `joinpath` instead of just `/`. It's also used in other parts of the code. Maybe we should refactor if we arrive at this as a general guideline?

Reply:
Yeah, we can keep `joinpath`. Just wanted to point it out.